pkg/operator/configobservation/cloudprovider/observe_cloudprovider_test.go

```diff
@@ -15,33 +15,54 @@ import (
 )
 
 func TestObserveCloudProviderNames(t *testing.T) {
-	indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
-	if err := indexer.Add(&corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cluster-config-v1",
-			Namespace: "kube-system",
-		},
-		Data: map[string]string{
-			"install-config": "platform:\n  aws: {}\n",
-		},
-	}); err != nil {
-		t.Fatal(err.Error())
-	}
-	listers := configobservation.Listers{
-		ConfigmapLister: corelistersv1.NewConfigMapLister(indexer),
-	}
-	result, errs := ObserveCloudProviderNames(listers, events.NewInMemoryRecorder("cloud"), map[string]interface{}{})
-	if len(errs) > 0 {
-		t.Fatal(errs)
-	}
-	cloudProvider, _, err := unstructured.NestedSlice(result, "extendedArguments", "cloud-provider")
-	if err != nil {
-		t.Fatal(err)
-	}
-	if e, a := 1, len(cloudProvider); e != a {
-		t.Fatalf("expected len(cloudProvider) == %d, got %d", e, a)
-	}
-	if e, a := "aws", cloudProvider[0]; e != a {
-		t.Errorf("expected cloud-provider=%s, got %s", e, a)
+	cases := []struct {
+		installConfig        string
+		cloudProvider        string
+		hasExtendedArguments bool
+	}{{
+		installConfig:        "platform:\n  aws: {}\n",
+		cloudProvider:        "aws",
+		hasExtendedArguments: true,
+	}, {
+		installConfig:        "platform:\n  libvirt: {}\n",
+		cloudProvider:        "",
```

Comment:
Given the "aws" test case, how come this is not "libvirt"? Perhaps I didn't look closely enough but I see for libvirt we return `return observerConfig, errs` - curious as to why "aws" is more explicit.

Reply:
This var is not being checked, as the func would immediately return `observedConfig` so in test `ObserveCloudProviderNames` would return immediately.

Reworked the test:
* `cloudProvider` is not initialized for none and libvirt
* test struct has `cloudProviderNum` - expected number of cloudproviders
* test now ensures expected and actual number of cloudproviders are equals

Reply:
I think `cloudProviderNum` is better expressed as `cloudProviderCount`. For me (at least) `Num` looks like an index.

Reply:
Right, I was choosing between these two, renamed to `Count`